lib/datasource/npm/__snapshots__/index.spec.ts.snap

```diff
@@ -641,10 +713,16 @@ Object {
   "registryUrl": "https://npm.mycustomregistry.com/",
   "releases": Array [
     Object {
+      "dependencies": undefined,
+      "devDependencies": undefined,
+      "gitRef": undefined,
       "releaseTimestamp": "2018-05-06T05:21:53.000Z",
       "version": "0.0.1",
     },
     Object {
+      "dependencies": undefined,
```

Comment:
what leads to these additional `undefined` fields which were empty previously?

Reply:
With the current implementation, `JSON.stringify({ foo: undefined }) === '{}'` with further parsing will remove the field.
I'm not sure if explicit undefined fields will do any harm after this merge, probably it can.
I see some decision space here:
1. Leave/remove explicit undefined fields
2. `undefined`, `null`, or something else for circular references (maybe, completely remove just like `undefined`)

Reply:
Are the above all circular reference fields? Should they happen in real scenarios or just under testing?